### ISSUE TITLE
How does the GUR module work in the paper

### ISSUE THREAD

**Issue by Twe1v0o** [C0ldstudy/Argus#4]
May I ask if you can solve some of my doubts ?
I noticed that in the training algorithm description of the paper, the hidden state of the previous snapshot of the RNN was not used. However, in the description section of the main text, it is mentioned that GRU will be used to capture topology and node changes in a series of DTG snapshots. 
At the same time, it seems that the code implementation does not encode the current subgraph based on the hidden state of the previous snapshot between snapshots during training. Instead, all the trained snapshots are encoded at once and then input into GRU to complete an epoch.
`
for e in range(args.epochs):
        # Get loss and send backward
        model.train()
        st = time.time()
        zs = model.forward(TData.TRAIN)
        loss = model.loss_fn(zs, TData.TRAIN, nratio=args.nratio, device=device, encoder_name=args.encoder_name)
        loss.backward()
        opt.step()
        elapsed = time.time()-st
        times.append(elapsed)
        l = loss.sum()
        print('[%d] Loss %0.4f  %0.2fs' % (e, l.item(), elapsed))
`

`
def forward(self, mask_enum, include_h=False, h0=None, no_grad=False):
        futs = self.encode(mask_enum, no_grad)
        zs = []
        zs, h0 = self.rnn(futs.to(self.device), h0, include_h=True)
        self.len_from_each = [embed.size(0) for embed in zs]
        self.z_dim = zs.size(-1)
        if include_h:
            return zs, h0
        else:
            return zs
`

**Comment by C0ldstudy** [comment]
Thank you so much for pointing it out! The misleading code of the hidden state is fixed [here](https://github.com/C0ldstudy/Argus/blob/8e7394d02be781221c7355bd45cc5c88fdd0c859/models/argus.py#L332). We actually used the GRU to capture the topology information.
